### PR TITLE
Adds volumeattach extension to openstack compute v2

### PR DESCRIPTION
**Pull request by jtopjian**
This commit adds support for the `os-volume_attachments` nova v2 api extension.

### CODE REVIEW

File: openstack/compute/v2/extensions/volumeattach/requests.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+	"github.com/racker/perigee"
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack/compute/v2/servers"
+	"github.com/rackspace/gophercloud/pagination"
+)
+
+// CreateOptsExt adds a VolumeAttachment option to the base CreateOpts.
+type CreateOptsExt struct {
+	servers.CreateOptsBuilder
```
Comment (jrperritt) [inline openstack/compute/v2/extensions/volumeattach/requests.go]: `CreateOptsExt` is used to add extension parameters to the base `CreateOpts` from the `servers` package. After looking at the OpenStack Nova source code,  I couldn't find where a volume could be attached to a server at creation time. Were you able to get this to work for you?

**Comment by jrperritt** [comment]
Looking good. Can you add a file with acceptance tests ([here](https://github.com/rackspace/gophercloud/tree/master/acceptance/openstack/compute/v2)) as well?

File: openstack/compute/v2/extensions/volumeattach/requests.go (hunk 1 above)
Comment (jtopjian) [inline openstack/compute/v2/extensions/volumeattach/requests.go]: You're right... I didn't actually check this.

**Comment by jtopjian** [comment]
I'll give it a shot. I was hoping I could squeak by without it ;)

**Comment by jtopjian** [comment]
How can I run the acceptance test for _just_ the file I create (and dependent utilities)

**Comment by jrperritt** [comment]
You can run a specific file by passing part of the test function name to the `run` flag. For example, if the test function is named TestAttachVolume, you might run those tests like so:
```
go test -tags 'acceptance fixtures' -run "AttachVolume" -v github.com/rackspace/gophercloud/acceptance/openstack/compute/v2
```

**Comment by jtopjian** [comment]
Thanks!

OK, got an acceptance test working.

File: acceptance/openstack/compute/v2/volumeattach_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+	volume, err := createVAVolume(t, blockClient)
+	if err != nil {
+		t.Fatalf("Unable to create volume: %v", err)
+	}
+
+	createVolumeAttachment(t, computeClient, blockClient, server.ID, volume.ID)
+
+	defer func() {
+		servers.Delete(computeClient, server.ID)
```
Comment (jrperritt) [inline acceptance/openstack/compute/v2/volumeattach_test.go]: You'll usually want to put `defer` statements immediately after the error checking lines for the same resource. The deferred function will be called even if the function errors out, but it has to called before then. For example, you'll want to move this function to delete the created server to line 104. Similarly for the function to delete the created volume.

Comment (jtopjian) [inline acceptance/openstack/compute/v2/volumeattach_test.go]: OK, that makes sense.

**Comment by jrperritt** [comment]
:+1: Great job, @jtopjian . Thank you for the PR and for your patience  :)

**Comment by jtopjian** [comment]
Pleasure was all mine. Thank you for all your help!

**Pull request merged**